src/app/components/ManualEntry.svelte

```diff
@@ -34,33 +34,47 @@
 
   const validateTableData = (name: string, wordlist: WordList): boolean => {
     const validTitle = validInput(name);
-
     const validRows = wordlist.every((wordAndCount: WordAndCount) => {
       return validInput(wordAndCount[WordAndCountInd.WORD]);
     });
     return validTitle && validRows;
   };
 
   const deleteRow = (i: number): void => {
-    rowDataFromManualEntry.splice(i, 1);
-    rowDataFromManualEntry = rowDataFromManualEntry;
+    wordlist.splice(i, 1);
+    wordlist = wordlist;
   };
 
-  const addNewRow = (): void => {
+  const addRow = (): void => {
     const newEntry: WordAndCount = ["", 0];
-    rowDataFromManualEntry.push(newEntry);
-    rowDataFromManualEntry = rowDataFromManualEntry;
+    wordlist.push(newEntry);
+    wordlist = wordlist;
```

Comment:
What are your thoughts on writing the update like this?:
```suggestion
    wordlist = [...wordlist, newEntry];
```

Reply:
Oh yes, I actually tried doing it that way and it didn't work! This would update `wordlist`, but not `tableData.wordlist` if I log their values. I looked into their documentation and it is in fact one of the solutions they suggest: https://svelte.dev/tutorial/updating-arrays-and-objects. But still can't seem to figure out why this approach doesn't work whereas the current approach oddly works. Do you happen to get it to work?

Reply:
Good catch, I had to tweak with the code a bit in order to get something like the above snippet to work :+1:.

What I did was remove the `wordlist` state and worked exclusively with `tableData.wordlist`. My guess is that state somehow gets duplicated when we both have `wordlist` and `tableData.wordlist`.

It might be worth considering removing `wordlist` to prevent these kinds of possible duplicate state bugs from cropping up in the future. What are your thoughts?

```suggestion
    tableData.wordlist = [...tableData.wordlist, newEntry];
```

Reply:
You're right. I'm also thinking someone else would bump into this later and try to do the same in the future, I might just do that!